### PR TITLE
[AIRFLOW-4900] Resolve incompatible version of Werkzeug

### PR DESCRIPTION
**Pull request by umanium**
Make sure you have checked _all_ steps below.

### Jira

- [x] https://issues.apache.org/jira/browse/AIRFLOW-4900

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
This PR Pin Werkzeug version to >= 0.15 as flask requirement needs it.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
No test added, because this is only changing requirements.

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by akki** [comment]
Thanks for the quick PR :)

I have a concern though:
Maybe changing `'flask>=1.0, <2.0',` to `'flask>=1.0, <1.1',` would be the more (backward) compatible change (Just in case upgrading to `flask==1.1` might have more incompatible issues with Airflow).
Also, in future, the same thing might happen when `flask==1.2` is released. 😕

**Comment by akki** [comment]
One more thing - I am not familiar with Airflow's backporting policies but since this issue affects Airflow `1.10.3` as well so it would be good to port this to other relevant branches.

File: setup.py [hunk 1]
```diff
@@ -370,7 +370,7 @@ def do_setup():
             'thrift>=0.9.2',
             'tzlocal>=1.4',
             'unicodecsv>=0.14.1',
-            'werkzeug>=0.14.1, <0.15.0',
+            'werkzeug>=0.15.0',
```
Comment (ashb) [inline setup.py]: I think let's just remove this entirely - we don't use werkzeug directly.

Comment (potiuk) [reply setup.py]: Agree. It does not seem to be used

**Review by ashb — APPROVED** [review]

**Review by mik-laj — APPROVED** [review]

**Comment by umanium** [comment]
Hi, thank you for resolving this 🙇 I'm sorry I was not with a computer after creating the PR.

**Comment by potiuk** [comment]
No worries! Thanks for looking into it and willingness to help :).

**Comment by techalchemy** [comment]
sorry to jump in -- y'all should consider adopting a tool that tells you which packages are interdependent so you can avoid strictly pinning things and causing this type of issue, not sure if poetry offers that but in pipenv, `pipenv graph` displays that information

**Pull request merged**